src/src/Helpers/Dallas1WireHelper.cpp

```diff
@@ -844,40 +849,40 @@ uint8_t Dallas_read(int8_t gpio_pin)
 /*********************************************************************************************\
 *  Dallas Write byte
 \*********************************************************************************************/
-void Dallas_write(uint8_t ByteToWrite, int8_t gpio_pin)
+void Dallas_write(uint8_t ByteToWrite, int8_t gpio_pin_tx)
 {
   uint8_t bitMask;
 
   for (bitMask = 0x01; bitMask; bitMask <<= 1) {
-    Dallas_write_bit((bitMask & ByteToWrite) ? 1 : 0, gpio_pin);
+    Dallas_write_bit((bitMask & ByteToWrite) ? 1 : 0, gpio_pin_tx);
   }
 }
 
 /*********************************************************************************************\
 *  Dallas Read bit
 \*********************************************************************************************/
-uint8_t Dallas_read_bit(int8_t gpio_pin)
+uint8_t Dallas_read_bit(int8_t gpio_pin_rx, int8_t gpio_pin_tx)
 {
-  if (gpio_pin == -1) { return 0; }
+  if (gpio_pin_rx == -1) { return 0; }
+  if (gpio_pin_tx == -1) { return 0; }
   uint8_t r;
   unsigned long start = micros();
 
     #if defined(ESP32)
   ESP32noInterrupts();
     #endif // if defined(ESP32)
-  digitalWrite(gpio_pin, LOW);
-  pinMode(gpio_pin, OUTPUT);
+  digitalWrite(gpio_pin_tx, LOW);
 
   while (usecPassedSince(start) < 6) {
     // Wait for 6 usec
   }
   unsigned long startwait = micros();
-  pinMode(gpio_pin, INPUT); // let pin float, pull up will raise
```

Comment:
The same here.
The pin MUST be floating or else the 1-pin config will no longer work.

Reply:
>  The pin MUST be floating or else the 1-pin config will no longer work.

Of course, that's what i already said in my initial comment:

> This is of course not a permanent solution, because it breaks the regular dallas 1-wire functionality. 

Tasmota fixes this by allowing the DS18B20 library to either run in single or dual mode:
https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_05_ds18x20.ino#L81